python/ray/experimental/channel/common.py

```diff
@@ -261,8 +285,14 @@ def __init__(self, input_channels: List[ChannelInterface]):
     def start(self):
         pass
 
-    def _read_list(self) -> Any:
-        return [c.read() for c in self._input_channels]
+    def _read_list(self, timeout: Optional[float] = None) -> List[Any]:
+        results = []
+        for c in self._input_channels:
+            start_time = time.monotonic()
+            results.append(c.read(timeout))
+            if timeout is not None:
+                timeout -= time.monotonic() - start_time
```

Comment:
I think we should take the max of this and 0.
```suggestion
                timeout -= time.monotonic() - start_time
                timeout = max(timeout, 0)
```

Reply:
This is better in case the other channels are already ready and would not block. Then we should just read from them and return, even though we're a bit over the timeout.